src/app/campaign/store/selectors/availability-allocation-actuals.selectors.ts

```diff
@@ -1,9 +1,141 @@
 import { createSelector } from '@ngrx/store';
-import { Availability, AvailabilityWeeklyRollup, AvailabilityRollup } from '../models';
+import {
+  Availability,
+  AvailabilityWeeklyRollup,
+  AvailabilityRollup,
+  AllocationPreview,
+  AvailabilityParams,
+  InventoryNumbers,
+  AvailabilityDay
+} from '../models';
 import { selectRoutedFlightAvailabilityEntities } from './availability.selectors';
 import { selectRoutedFlightAllocationPreviewEntities } from './allocation-preview.selectors';
 import { selectRoutedFlightId, selectRoutedLocalFlightZones } from './flight.selectors';
 
+export const getAllocationPreviewForDate = (
+  dateString: string,
+  zone: string,
+  allocationPreviewEntities: { [id: string]: AllocationPreview }
+): number => {
+  if (allocationPreviewEntities && allocationPreviewEntities[`${zone}_${dateString}`]) {
+    return allocationPreviewEntities[`${zone}_${dateString}`].goalCount;
+  }
+};
+
+export const getMidnightUTC = (date: Date): number => {
+  return date.valueOf() - (date.valueOf() % (24 * 60 * 60 * 1000));
+};
+
+export const getEndOfWeek = (date: Date): Date => {
```

Comment:
yeah... so there's a mix of Date and Moment in this because of the timing on the whole thing of adding moment into the project while I was in progress on this ticket. Moment is a bit less heavy lifting with its UTC mode.